pkg/3scale/amp/component/s3.go

```diff
@@ -140,6 +147,8 @@ func (s3 *S3) AddS3PostprocessOptionsToSystemEnvironmentCfgMap(objects []common.
 	systemEnvCfgMap.Data["FILE_UPLOAD_STORAGE"] = "s3"
 	systemEnvCfgMap.Data["AWS_BUCKET"] = s3.Options.awsBucket
 	systemEnvCfgMap.Data["AWS_REGION"] = s3.Options.awsRegion
+	systemEnvCfgMap.Data[StorageServiceEndpointScheme] = s3.Options.storageServiceEndpointScheme
```

Comment:
Notice how this is in the method AddS3PostprocessOptionsToSystemEnvironmentCfgMap.
This is a postprocess method of S3 for the templates.
Be aware the configuration of the envvars and setting of the valus in the system-environment ConfigMap in the templates use this method but NOT in the operator (Postprocess methods are specific and needed for the templates)